gson/src/main/java/com/google/gson/stream/JsonWriter.java

```diff
@@ -160,6 +166,9 @@
     HTML_SAFE_REPLACEMENT_CHARS['&'] = "\\u0026";
     HTML_SAFE_REPLACEMENT_CHARS['='] = "\\u003d";
     HTML_SAFE_REPLACEMENT_CHARS['\''] = "\\u0027";
+
+    // Syntax as defined by https://datatracker.ietf.org/doc/html/rfc8259#section-6
+    VALID_JSON_NUMBER_PATTERN = Pattern.compile("-?(?:0|[1-9][0-9]*(?:\\.[0-9]+)?(?:[eE][-+]?[0-9]+)?)");
```

Comment:
Google's tests found a problem with this: the string `0.8252564` is rejected. I think it should be:
`-?(?:0|[1-9][0-9]*)(?:\\.[0-9]+)?(?:[eE][-+]?[0-9]+)?`
I removed the outermost group because it doesn't seem necessary.

Reply:
Also, could you just make this an initializer on `VALID_JSON_NUMBER_PATTERN`? Unlike the other fields here it's just a plain initialization. You might want to move it before those other fields so they remain next to the `static {}` block that initializes them.

Reply:
You are right, the pattern was incorrect, thanks a lot for noticing this! I have fixed it and also added more tests for this.